File: app/src/main/java/uz/revolution/mantiq_qani/fragment/ScoresFragment.java
package uz.revolution.mantiq_qani.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import uz.revolution.mantiq_qani.GameActivity;
import uz.revolution.mantiq_qani.R;
import uz.revolution.mantiq_qani.adapter.FragmentAdapter;
import uz.revolution.mantiq_qani.dataBase.Database;
import uz.revolution.mantiq_qani.model.MyData;

import java.util.ArrayList;

public class ScoresFragment extends Fragment {
    private ListView listView;
    private ArrayList<MyData> data;
    private FragmentAdapter adapter;
    private Database database = Database.getDatabase();
    private Button cancel;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        View view = LayoutInflater.from(getContext()).inflate(R.layout.list_fragment, container, false);
        return view;

    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        listView = view.findViewById(R.id.listview);
        cancel = view.findViewById(R.id.scores_cancel_btn);
        loadData();
        adapter = new FragmentAdapter(data);
        listView.setAdapter(adapter);

        cancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((GameActivity) getActivity()).getSupportFragmentManager().popBackStack();
            }
        });
    }

    private void loadData() {
        data = new ArrayList<>();
        data = database.getUsersData();
    }
}

File: app/src/main/java/uz/revolution/mantiq_qani/fragment/ScoresFragment.kt
package uz.revolution.mantiq_qani.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import uz.revolution.mantiq_qani.adapter.FragmentAdapter
import uz.revolution.mantiq_qani.dataBase.MySharedPreferences
import uz.revolution.mantiq_qani.databinding.ListFragmentBinding
import uz.revolution.mantiq_qani.model.MyData
import java.util.*

class ScoresFragment : Fragment() {
    private var data: ArrayList<MyData>? = null
    private var adapter: FragmentAdapter? = null
    lateinit var binding: ListFragmentBinding
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = ListFragmentBinding.inflate(layoutInflater, container, false)

        val database = MySharedPreferences.getUsersDataList()
        adapter = database.let { FragmentAdapter(it) }
        binding.listview.adapter = adapter

        binding.scoresCancelBtn.setOnClickListener {
            findNavController().popBackStack()
        }

        return binding.root
    }
}